roles/solr_collection/tasks/main.yml

```diff
@@ -36,6 +40,36 @@
         url: "{{ solr_url }}admin/collections?action=CREATE&name={{ solr_collection }}&collection.configName={{ solr_configset }}&numShards={{ num_shards }}&replicationFactor={{ replication_factor }}&maxShardsPerNode={{ shards_per_node }}"
       when: solr_collection not in solr_collections_list.json.collections
 
+    - name: Check Solr collection status if it previously existed
+      uri:
+        url: "{{ solr_url }}admin/collections?action=COLSTATUS&collection={{ solr_collection }}&coreInfo=true"
+      register: solr_collection_status
+      when: solr_collection in solr_collections_list.json.collections
+
+    - name: Get collection information from collection status response
+      # get collection-specific portion of the collection status response
+      set_fact:
+        solr_collection_info: "{{ solr_collection_status.json[ solr_collection ]}}"
+      when: solr_collection_status is defined
+
+    # NOTE: number of shards can't be changed via modify collection;
+    # must use reindex; warning, this may be slow
+    - name: Modify Solr collection number of shards if changed
+      uri:
+        url: "{{ solr_url }}admin/collections?action=REINDEXCOLLECTION&name={{ solr_collection }}&numShards={{ num_shards }}"
+      when: solr_collection_info["activeShards"] != num_shards
+      # is it ok to use active shards total and ignore inactive?
```

Comment:
I'm not sure what an inactive shard is. Interested if you find out more info on this question.